sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -749,9 +749,11 @@ case class SubqueryExec(name: String, child: SparkPlan)
   private lazy val relationFuture: Future[Array[InternalRow]] = {
     // relationFuture is used in "doExecute". Therefore we can get the execution id correctly here.
     val executionId = sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)
+    val localProps = Utils.cloneProperties(sparkContext.getLocalProperties)
     Future {
       // This will run in another thread. Set the execution id so that we can connect these jobs
       // with the correct execution.
+      sparkContext.setLocalProperties(localProps)
```

Comment:
what's the consequence if we don't do this? Do you run a UDF in subquery which accesses task context properties?

Reply:
yes, if the subquery output depends on the task properties, it will affect i have added a UT case to simulate the behaviour